web/src/muted_topics_ui.js

```diff
@@ -19,7 +20,10 @@ export function rerender_for_muted_topic(old_muted_topics) {
     if (overlays.settings_open() && settings_muted_topics.loaded) {
         settings_muted_topics.populate_list();
     }
-
+    // Before re-rendering, we destroy all the tooltips. However, after re-rendering,
+    // there are no instances left, so the tooltips don't appear. To ensure that they
+    // appear, we need to initialize all the tooltips again.
+    tippyjs.initialize();
```

Comment:
Is this really required? I recall that Tippyjs just creates new instances if there are none present, as long the trigger for them is still active.

Reply:
Yes, it is necessary because the tooltips appear due to the active triggers and instances. If we remove all the instances, the triggers will no longer be active, and thus the tooltips will not appear. Therefore, we need to initialize the tooltips again to make them appear. I tested the code changes several times to disprove this reasoning by commenting out the initialization call, but the tooltips did not appear. Please let me know if you have any other thoughts. Additionally, I changed the location of the initialize function call. Since we need to call it every time we destroy the tooltips, it would be better to call it inside the `destroyAllInstances` after removing all instances.

Reply:
Ok, let's do this
```
    for (const instance of messageListTippyInstances) {
        instance.hide();
    }
```
instead of `instance.destroy` since we only care about them being hidden. I don't think you will need to call `initialize` then.

Reply:
I tried your suggested approach, but it didn't work. We have to destroy the tooltips to prevent their second appearance and then initialize them again. I have attached a demo of your approach.

![error-](https://user-images.githubusercontent.com/66828942/233780897-90395cb2-b98f-461f-940b-2df776e7670b.gif)

Reply:
oh! I think we are destroying the hook that is set by `delegate('body'` here. So something like this should work:
```
    for (const instance of message_list_tippy_instances) {
        if (instance.reference === document.body) {
            continue;
        }
        instance.destroy();
    }
```

Reply:
That definitely worked. Thank you for the suggestion.